src/DBFTables.jl

```diff
@@ -308,4 +353,117 @@ function Base.getproperty(dbf::Table, name::Symbol)
+    rsize = UInt16(sum(x -> x.length, fields)) + 1
+
+    version = 0x03
+    last_update = today()
+    incomplete = false
+    encrypted = false
+    mdx = false
+    lang_id = 0x00
+
+    h = Header(version, last_update, records, hsize, rsize, incomplete, encrypted, mdx, lang_id, fields, fieldcolumns)
+    out = Base.write(io, h)
+
+    for row in Tables.rows(tbl)
+        out += write_record(io, fields, row)
+    end
+    out += Base.write(io, 0x1a)  # EOF marker
+    return out
+end
+
+function get_field_descriptors(tbl)
+    fields = FieldDescriptor[]
+    dct = Tables.dictcolumntable(tbl)
+    sch = Tables.schema(dct)
+    for (name, type) in zip(sch.names, sch.types)
+        ndec = 0x0
+        len = 0x0
+        dbf_type = 'C'
+        T = Base.nonmissingtype(type)
+        if T isa Date
+            dbf_type = 'D'
+            len = 0x08
+        elseif T <: AbstractString
+            # TODO: support memos.  Currently strings > 254 bytes will error
+            len = UInt8(maximum(x -> length(string(x)), dct[name]))
+            if len > 254
+                @warn "Strings will be truncated to 254 characters."
+                len = 254
+            end
+            dbf_type = 'C'
+        elseif type === Float64
+            dbf_type = 'O'
+            len = 0x08
+            ndec = 0x01
+        elseif T <: AbstractFloat
+            dbf_type = 'F'
+            len = UInt8(20)
+            ndec = 0x01
+        elseif T <: Bool
+            dbf_type = 'L'
+            len = 0x1
+        elseif T <: Date
+            dbf_type = 'D'
+            len = 0x8
+        elseif T <: Integer
+            dbf_type = 'N'
+            len = UInt8(maximum(x -> length(string(x)), dct[name]))
+        elseif T <: AbstractFloat
+            dbf_type = 'N'
+            ndec = 0x01
+            len = UInt8(maximum(x -> length(string(x)), dct[name]))
+        else
+            @warn "Field $name has no known matching DBF data type for $T.  Data will be stored as the DBF character data type ('C')."
+            len = UInt8(maximum(x -> length(string(x)), dct[name]))
+        end
+        push!(fields, FieldDescriptor(name, type, dbf_type, len, ndec))
+    end
+    fields, UInt32(length(first(dct)))
+end
+
+function write_record(io::IO, fd::Vector{FieldDescriptor}, row)
+    out = 0
+    out += Base.write(io, ' ')  # deletion marker ' '=valid, '*'=deleted
+    for (field, val) in zip(fd, row)
+        out += Base.write(io, _val(field, val))
+    end
+    return out
+end
+
+function _val(field::FieldDescriptor, val)::Union{String, Float64}
+    char = field.dbf_type
+    if char == 'L'
+        ismissing(val) && return "?"
+        val ? "T" : "F"
+    elseif ismissing(val)
+        ' ' ^ field.length
+    elseif char == 'C'
+        rpad(val, field.length)
+    elseif char == 'D'
+        Dates.format(val, "yyyymmdd")
+    elseif char == 'O'
+        val  # <-- the Float64 return value
+    elseif char == 'F'
+        # TODO: fix for scientific notation
```

Comment:
was wondering about scientific notation too

Reply:
I added that note before implementing the `'O'` data type for `Float64`.  Only `Float32`s and smaller will end up using `'F'` and they all fit in the 20-character width.  I was thinking the scientific notation was an issue for Float64 because the string length could end up being 20+ characters.